Libraries/Image/RCTImageLoader.m

```diff
@@ -480,6 +470,18 @@ - (RCTImageLoaderCancellationBlock)loadImageWithURLRequest:(NSURLRequest *)image
   __block void(^cancelLoad)(void) = nil;
   __weak RCTImageLoader *weakSelf = self;
 
+  // Load handlers are executed before the generic caching logic so they are responsible
+  // for caching the images if necessary.
+  id<RCTImageURLLoader> loadHandler = [self imageURLLoaderForURL:imageURLRequest.URL];
```

Comment:
We probably need another approach. To clarify: your issue is that we first have to schedule this on the `_URLCacheQueue` and then we go to the main queue right? Keep in mind that we also do this for memory reasons, if you start loading a ton of images on the main thread, skipping our queueing, you'll run into OOMs as well.